sdk/keyvault/azure-security-keyvault-keys/src/main/java/com/azure/security/keyvault/keys/cryptography/CryptographyAsyncClient.java

```diff
@@ -924,55 +896,65 @@ public Mono<SignResult> signData(SignatureAlgorithm algorithm, byte[] data) {
     @ServiceMethod(returns = ReturnType.SINGLE)
     public Mono<VerifyResult> verifyData(SignatureAlgorithm algorithm, byte[] data, byte[] signature) {
         try {
-            return withContext(context -> isValidKeyLocallyAvailable().flatMap(available -> {
-                if (!available) {
-                    return implClient.verifyDataAsync(algorithm, data, signature, context);
-                }
+            return withContext(context -> isLocalClientAvailable().flatMap(available -> {
+                if (available) {
+                    verifyKeyPermissions(jsonWebKey, KeyOperation.VERIFY);
 
-                if (!checkKeyPermissions(key.getKeyOps(), KeyOperation.VERIFY)) {
-                    return Mono.error(LOGGER.logExceptionAsError(new UnsupportedOperationException(
-                        "Verify operation is not allowed for key with id: " + key.getId())));
+                    return localKeyCryptographyClient.verifyDataAsync(algorithm, data, signature, context);
+                } else {
+                    return implClient.verifyDataAsync(algorithm, data, signature, context);
                 }
-
-                return localKeyCryptographyClient.verifyDataAsync(algorithm, data, signature, key, context);
             }));
         } catch (RuntimeException ex) {
             return monoError(LOGGER, ex);
         }
     }
 
-    private Mono<Boolean> isValidKeyLocallyAvailable() {
-        if (localOperationNotSupported) {
-            return Mono.just(false);
+    private Mono<Boolean> isLocalClientAvailable() {
+        if (!attemptedToInitializeLocalClient) {
+            return retrieveJwkAndInitializeLocalAsyncClient()
+                .map(localClient -> {
+                    localKeyCryptographyClient = localClient;
+                    jsonWebKey = localKeyCryptographyClient.getJsonWebKey();
+                    attemptedToInitializeLocalClient = true;
+
+                    return true;
+                })
+                .onErrorResume(e -> {
```

Comment:
Are we going to fall back to service-side operations even if this error is transient?

Reply:
We'll set the `attemptedToInitializeLocalClient` flag to true only after initializing a local client, so if the process fails at any point before that we will try again when attempting another cryptographic operation.

Reply:
I'll update the message to show that we're only deferring the current operation to the service, not all of them.

Reply:
Should we log that on INFO or DEBUG?